### PR TITLE
1.x Encode @class entry in InstanceInfo.metadata for backward compatibility

### PR DESCRIPTION
**Pull request by tbak**

### CODE REVIEW

File: eureka-client/src/main/java/com/netflix/discovery/converters/jackson/InstanceInfoJsonBeanSerializer.java [hunk 1]
```diff
@@ -29,8 +32,18 @@
  * @author Tomasz Bak
  */
 class InstanceInfoJsonBeanSerializer extends BeanSerializer {
-    InstanceInfoJsonBeanSerializer(BeanSerializerBase src) {
+
+    private static final Map<String, String> EMPTY_MAP = Collections.singletonMap("@class", "java.util.Collections$EmptyMap");
```
Comment (qiangdavidliu) [inline eureka-client/src/main/java/com/netflix/discovery/converters/jackson/InstanceInfoJsonBeanSerializer.java]: add a comment about why this is necessary?

Comment (qiangdavidliu) [inline eureka-client/src/main/java/com/netflix/discovery/converters/jackson/InstanceInfoJsonBeanSerializer.java]: nvm saw it was commented on later.

File: eureka-client/src/main/java/com/netflix/discovery/converters/jackson/EurekaJsonJacksonCodec.java [hunk 2]
```diff
@@ -37,7 +39,7 @@ public EurekaJsonJacksonCodec() {
     public EurekaJsonJacksonCodec(final KeyFormatter keyFormatter, boolean compact) {
```
Comment (qiangdavidliu) [inline eureka-client/src/main/java/com/netflix/discovery/converters/jackson/EurekaJsonJacksonCodec.java]: I can imaging potential future cases where we may have multiple different compact types. As this is exposed as public API, can we change this from a boolean to a more flexible type?

Comment (tbak) [inline eureka-client/src/main/java/com/netflix/discovery/converters/jackson/EurekaJsonJacksonCodec.java]: I wouldn't say this is public API. All codec stuff is internal
implementation until we provide "official" codec API.

Comment (qiangdavidliu) [inline eureka-client/src/main/java/com/netflix/discovery/converters/jackson/EurekaJsonJacksonCodec.java]: It's in a public constructor and there are already cases of extensions of the Codec classes.

**Comment by qiangdavidliu** [comment]
ship it.

**Pull request merged**